File: C/367.valid-perfect-square.c
bool isPerfectSquare(int num) {
    if (num < 1)
        return false;
    if (num == 1)
        return true;
    int i;
    for (i = 1; i < num; i++) {
        if (i * i == num)
            return true;
        if (i * i > num)
            return false;
    }
    return false;
}

File: Kotlin/367.valid-perfect-square.kt
class Solution {
    fun isPerfectSquare(num: Int): Boolean {
        if (num < 1) return false
        if (num == 1) return true
        for (i in 1..num) {
            if (i * i == num) return true
            if (i * i > num) return false
        }
        return false
    }
}